media/js/base/banners/m24-pencil-banner.es6.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+import { getConsentCookie } from '../consent/utils.es6';
+
+let _pencilBanner;
+const M24PencilBanner = {};
+const BANNER_ID = 'pencil-banner';
+
+M24PencilBanner.consentsToCookies = function () {
+    const cookie = getConsentCookie();
+
+    if (cookie) {
+        return cookie.preference;
+    } else {
+        return true;
+    }
+};
+
+M24PencilBanner.setCookie = function () {
+    /**
+     * When closing a banner we set a "preference cookie" to remember
+     * that a user no longer wishes to see it on repeat page visits.
+     * Legal are OK to set this cookie without explicit consent in
+     * EU/EAA countries because:
+     *
+     * 1) The cookie is not used for tracking purposes in any way.
+     * 2) The cookie is set only after an explicit action by the user
+     *    that signals clear intent.
+     *
+     * We still honor not setting this cookie if preference cookies
+     * have been explicitly rejected by the user.
+     */
+    if (!M24PencilBanner.consentsToCookies()) {
+        return;
```

Comment:
**Enhancement:** (no priority / not blocking) If they've refused cookies we could hide the close button.

Reply:
I think I'm going to leave the close button as-is, so people can at least dismiss it on the page they are trying to read.